packages/retail-ui-extensions/src/components/List/List.ts

```diff
@@ -1,22 +1,76 @@
 import {createRemoteComponent} from '@remote-ui/core';
+import {TagProps} from 'components/Tag';
 
-export interface ListItem {
-  uuid: string;
-  title: string;
-  subtitle?: string;
-  subtitleColor?: 'subdued' | 'success' | 'warning' | 'critical';
-  imageUri?: string;
-  detail?: string;
-  detailColor?: 'subdued' | 'action';
-  onTap?: () => void;
+export interface ListRowLeftSide {
+  /**
+   * The main label that will be displayed on the left side of the row.
+   */
+  label: string;
+  /**
+   * The subtitles to display beneath the main label. Up to 3 subtitles can be displayed.
+   */
+  subtitle?: readonly [string, string?, string?];
+  /**
+   * A colored badge that is displayed underneath the `title` and `subtitles`.
+   */
+  badge: TagProps;
+  image?: {
+    /**
+     * A link to an image to be displayed on the far left side of the row.
+     */
+    source: string;
+  };
+}
+
+export interface ListRowRightSide {
+  /**
+   * The main label that will be displayed on the right side of the row.
+   */
+  label?: string;
+}
+
+export interface ListRow {
+  /**
+   * The unique identifier for this list item.
+   */
+  id: string;
+  /**
+   * The user interface of the left side of the row.
+   */
+  leftSide: ListRowLeftSide;
+  /**
+   * The user interface of the right side of the row.
+   */
+  rightSide?: ListRowRightSide;
+  /**
+   * Callback for when the user taps the row.
+   */
+  onPress?: () => void;
 }
 
 export interface ListProps {
-  data: ListItem[];
-  showDisclosureIndicator: boolean;
-  isLoadingMore?: boolean;
+  /**
+   * A large display title at the top of the `List`.
+   */
   title?: string;
+  /**
+   * The array of `ListItem` which will be converted into rows for this list.
```

Comment:
Do we still call this `ListItem`? Should this be `ListRow`?

```suggestion
   * The array of `ListRow` which will be converted into rows for this list.
```